collector-exoplayer/src/test/java/com/bitmovin/analytics/exoplayer/ExoPlayerAdapterTest.kt

```diff
@@ -35,13 +35,25 @@ class ExoPlayerAdapterTest {
     fun testNoStateTransitionToQualityChangeIfBitrateDidNotChange() {
         val bitrate = 3000
         `when`(stateMachine.currentState).thenReturn(PlayerState.PLAYING)
+        `when`(stateMachine.isQualityChangeEventEnabled).thenReturn(true)
         adapter.fakePosition = 20
         adapter.onDecoderInputFormatChanged(getEventTime(20L), 0, Format.createVideoSampleFormat(null, null, null, bitrate, 1, 300, 300, 64F, null, null))
         verify(stateMachine, times(1)).transitionState(eq(PlayerState.QUALITYCHANGE), ArgumentMatchers.anyLong())
+
         adapter.onDecoderInputFormatChanged(getEventTime(30L), 0, Format.createVideoSampleFormat(null, null, null, bitrate, 1, 300, 300, 64F, null, null))
         verify(stateMachine, times(1)).transitionState(eq(PlayerState.QUALITYCHANGE), ArgumentMatchers.anyLong())
```

Comment:
```suggestion
        verify(stateMachine, times(0)).transitionState(eq(PlayerState.QUALITYCHANGE), ArgumentMatchers.anyLong())
```
The name of the test suggests that we do not expect a second quality change event, is this intended?

Reply:
Hmm, `testNoStateTransitionToQualityChangeIfBitrateDidNotChange` works as it did previously... I only added mock for additional condition `isQualityChangeEventEnabled`. 
As far as I can see first call to state machine should be when `onDecoderInputFormatChanged` event is first time triggered (on line 40). Then on line 43 we trigger it again, with same parameters. Now because there has been no change to bitrate (parameters of both events are the same) state machine is not called again. I think that is why on line 44 we have check to confirm that only one call has been made. 🤔

Reply:
Ah got it. Mistakenly though that the `verify(..., times(N))` will be reset every time this is called 🤦. Thanks for clarifying :)